Tools/TestWebKitAPI/Tests/WebKitCocoa/CookieStoreAPI.mm

```diff
@@ -61,4 +61,26 @@ HTTPServer server({
     EXPECT_TRUE([result isEqualToString:@"cookieValue"]);
 }
 
+TEST(WebKit, CookieStoreSetCookieForPublicSuffixDomain)
+{
+    HTTPServer server({
+        { "/foo"_s, { "foo"_s } }
+    }, HTTPServer::Protocol::HttpsProxy);
+
+    RetainPtr configuration = server.httpsProxyConfiguration();
+    RetainPtr webView = adoptNS([[TestWKWebView alloc] initWithFrame:CGRectMake(0, 0, 300, 300) configuration:configuration.get()]);
+
+    RetainPtr navigationDelegate = adoptNS([TestNavigationDelegate new]);
+    [navigationDelegate allowAnyTLSCertificate];
+    [webView setNavigationDelegate:navigationDelegate.get()];
+
+    [webView loadRequest:[NSURLRequest requestWithURL:[NSURL URLWithString:@"https://example.com/foo"]]];
+    [navigationDelegate waitForDidFinishNavigation];
+
+    NSError *error;
+    [webView objectByCallingAsyncFunction:@"await cookieStore.set({ name: 'cookieName', value: 'cookieValue', domain: 'com' });" withArguments:nil error:&error];
```

Comment:
Can we add a test case for `co.uk` ?

Reply:
Added and uploaded! It's ok to have them both be in the same "CookieStoreSetCookieForPublicSuffixDomain" test, or should I make an entirely new one for it?